install.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+if [ -z "$BIN_DIR" ]; then
+  BIN_DIR=$(pwd)
+fi
+
+THE_ARCH_BIN=''
+THIS_PROJECT_OWNER='dunglas'
+THIS_PROJECT_NAME='frankenphp'
+
+THISOS=$(uname -s)
+ARCH=$(uname -m)
+
+case $THISOS in
+   Linux*)
+      case $ARCH in
+        arm64)
+          THE_ARCH_BIN=""
+          ;;
+        aarch64)
+          THE_ARCH_BIN="$THIS_PROJECT_NAME-linux-aarch64"
+          ;;
+        armv6l)
+          THE_ARCH_BIN=""
+          ;;
+        armv7l)
+          THE_ARCH_BIN=""
+          ;;
+        *)
+          THE_ARCH_BIN="$THIS_PROJECT_NAME-linux-x86_64"
+          ;;
+      esac
+      ;;
+   Darwin*)
+      case $ARCH in
+        arm64)
+          THE_ARCH_BIN="$THIS_PROJECT_NAME-mac-arm64"
+          ;;
+        *)
+          THE_ARCH_BIN="$THIS_PROJECT_NAME-mac-x86_64"
+          ;;
+      esac
+      ;;
+   Windows|MINGW64_NT*)
+        THE_ARCH_BIN=""
+      ;;
+esac
+
+if [ -z "$THE_ARCH_BIN" ]; then
+   echo "This script is not supported on $THISOS and $ARCH"
+   exit 1
+fi
+
+curl -kL --progress-bar https://github.com/$THIS_PROJECT_OWNER/$THIS_PROJECT_NAME/releases/latest/download/$THE_ARCH_BIN -o $BIN_DIR/$THIS_PROJECT_NAME
```

Comment:
Sorry for coming back around, I should have thought about this earlier: one downside to downloading directly to the installed location is that if you are downloading an updated version and replacing an old version, and the download fails, you could end up in a broken state until retried. It's probably a non-issue, but just something to consider.

Reply:
Actually this script is battle tested on https://github.com/kevincobain2000/gobrew/blob/master/git.io.sh, I never had issues raised with regards to failed download.

Reply:
Heh, my ISP breaks routing pretty regularly to certain parts of the internet (mainly Azure ip's). It's a thing that causes serious issues with downloads sometimes (the route to Azure from my location is VERY asymmetric). 

Anywho, the issue is probably a non-issue because people can just retry until it works. For example, I don't usually report these issues to web hosts but have to go to an obscure forum and report that routing is broken (again).